lib/sendf.c

```diff
@@ -254,6 +334,15 @@ ssize_t Curl_send_plain(struct connectdata *conn, int num,
                         const void *mem, size_t len, CURLcode *code)
 {
   curl_socket_t sockfd = conn->sock[num];
+#ifdef USE_WINSOCK_WORKAROUNDS
+  /* WinSock will destroy unread received data if send() is
+     failed.
+     To avoid lossage of received data, recv() must be
+     performed before every send() if any incoming data is
+     available. */
+    pre_receive_plain(conn, num);
+#endif /* _WIN32 */
```

Comment:
https://github.com/curl/curl/blob/master/docs/CODE_STYLE.md#platform-dependent-code

It is nicer if you define pre_receive_plain() to be an empty macro for platforms that don't need the function, as then you don't need any #ifdef at all in this function. This pattern can be used on more places to make the code easier on the eye (== less #ifdefs).